### ISSUE TITLE
Issue with Docker Socket

### ISSUE THREAD

**Issue by XzMrtrevorzX** [goauthentik/authentik#2779]
**Describe your question/**
Trying to see if others have this same error

**Relevant infos**
Docker version 20.10.5, build 55c4c88

**Screenshots**
N/A

**Logs**
docker: Error response from daemon: create /var/run/docker.sock: " /var/run/docker.sock" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.

**Version and Deployment (please complete the following information):**
 - authentik version: authentik:2022.4.1
 - Deployment: Docker

**Additional context**
Just wanted to know if there was a workaround for this issue.

**Comment by RoboMagus** [comment]
What's your compose config for running this container?

If you've copied this error message straight from your logs, it seems that you have included a ' ' (space) in your path and that would be causing the issue ;)

**Comment by XzMrtrevorzX** [comment]
Not sure on that. using Unraid with Docker

**Comment by gueldenstone** [comment]
Is the docker socket really needed? I could not find any documentation on what the worker does with it? Maybe someone could shed some light on that?

**Comment by RoboMagus** [comment]
It's not explicitly mentioned indeed, but if you deploy based on the [provided docker-compose example](https://raw.githubusercontent.com/goauthentik/authentik/version-2022.4/docker-compose.yml) you'll see it mentioned on the worker node.

**Comment by gueldenstone** [comment]
Thanks, yes, thats why I came across it. I found out that it's used to spawn outposts if I'm not mistaken. Can someone confirm? So unless one would like to use other outposts than the embedded the mounting of the docker socket is necessary, otherwise not? Sorry, I'm quite new to authentik and just trying to figure stuff out.

**Comment by RoboMagus** [comment]
@gueldenstone, You got me curious as well, as I've just copied the relevant parts of the docker-compose example into my config and was satisfied once I got it all up and running.

My config only contains the embedded outpost, so by your reasoning I should not need to have the docker socket mapped.
I've just modified my config to exclude the mapping of the docker socket and given it a go. After restarting both the server and worker nodes nothing seems different. All my login flows still work just fine!

With the exception of the following warning being logged: `DockerException(\"Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))\")`, nothing seems to have changed.

**Comment by gueldenstone** [comment]
That's what I figured. Because I could not find out what the worker is doing with the socket, I hesitated to mount it. I could not tell if anything is not working because I'm not really using much. But it seems you can just skip it. Should put that in the [documentation](https://goauthentik.io/docs/installation/docker-compose)

**Comment by BeryJu** [comment]
The original error seems to indeed happen due to a leading space in the path.

The reason for the docker socket is documented here https://goauthentik.io/docs/outposts/integrations/docker (and yes I agree it's not linked clearly, I had to search for the page too)

TL;DR it's used if you want to create additional outposts and have authentik manage them. For most setups this probably won't be needed as the embedded outpost can be used.

**Comment by septatrix** [comment]
> The reason for the docker socket is documented here https://goauthentik.io/docs/outposts/integrations/docker (and yes I agree it's not linked clearly, I had to search for the page too)

Could this be added to the `docker-compose.yaml` file? I too was very confused why it was required there and the existing comments do not explain it well.
